ios/Exponent/Versioned/Core/UniversalModules/EXNotifications/EXScopedNotificationCategoryMigrator.m

```diff
@@ -0,0 +1,71 @@
+// Copyright 2021-present 650 Industries. All rights reserved.
+
+#import "EXScopedNotificationCategoryMigrator.h"
+#import "EXScopedNotificationsUtils.h"
+
+@implementation EXScopedNotificationCategoryMigrator
+
++ (void)migrateLegacyScopedCategoryIdentifiersForProject:(NSString *)experienceId
+{
+  NSString *prefixToReplace = [NSString stringWithFormat:@"%@-", experienceId];
+  [EXScopedNotificationCategoryMigrator renameCategoryIdentifiersWithPrefix:prefixToReplace withBlock:^(UNNotificationCategory *oldCategory) {
+    NSString *unscopedLegacyCategoryId = [EXScopedNotificationsUtils unscopedLegacyCategoryIdentifierWithId:oldCategory.identifier forExperience:experienceId];
+    NSString *newCategoryId = [EXScopedNotificationsUtils scopedCategoryIdentifierWithId:unscopedLegacyCategoryId forExperience:experienceId];
+    UNNotificationCategory *newCategory = [EXScopedNotificationCategoryMigrator createNewCategoryFrom:oldCategory withNewIdentifier:newCategoryId];
+    return newCategory;
+  }];
+}
+
++ (void)unscopeLegacyCategoryIdentifiersForProject:(NSString *)experienceId
+{
+  NSString *prefixToReplace = [NSString stringWithFormat:@"%@-", experienceId];
+  [EXScopedNotificationCategoryMigrator renameCategoryIdentifiersWithPrefix:prefixToReplace withBlock:^(UNNotificationCategory *oldCategory) {
+    NSString *unscopedCategoryId = [EXScopedNotificationsUtils unscopedLegacyCategoryIdentifierWithId:oldCategory.identifier forExperience:experienceId];
+    UNNotificationCategory *newCategory = [EXScopedNotificationCategoryMigrator createNewCategoryFrom:oldCategory withNewIdentifier:unscopedCategoryId];
+    return newCategory;
+  }];
+}
+
++ (void)renameCategoryIdentifiersWithPrefix:(NSString *)oldPrefix withBlock:(UNNotificationCategory *(^)(UNNotificationCategory *category))renameCategoryBlock
```

Comment:
for clarity that this doesn't apply to non-legacy scoped identifiers, which we can cleanly parse:

```suggestion
+ (void)renameLegacyCategoryIdentifiersWithPrefix:(NSString *)oldPrefix withBlock:(UNNotificationCategory *(^)(UNNotificationCategory *category))renameCategoryBlock
```